tests/runtime/test_service_factory.py

```diff
@@ -165,3 +167,130 @@ def test_get_service_proto_module():
         "caikit_runtime_pb2", mock_config
     )
     assert service_proto_module == caikit_runtime_pb2
+
+
+# Local
+import sample_lib
+
+MODULE_LIST = [
+    module_class
+    for module_class in caikit.core.MODULE_REGISTRY.values()
+    if module_class.__module__.partition(".")[0] == "sample_lib"
+]
+
+### Test ServicePackageFactory._remove_exclusions_from_module_list
+def test_get_and_filter_modules_respects_excluded_task_type():
+    assert len(MODULE_LIST) == 6  # there are 6 modules in sample_lib
+    with temp_config_parser(
+        {"service_generation": {"task_types": {"excluded": ["sample_task"]}}}
+    ) as cfg:
+        clean_modules = ServicePackageFactory._get_and_filter_modules(cfg, "sample_lib")
+        assert len(clean_modules) == 1
+        assert "sample_task" not in str(clean_modules)
+
+
+def test_get_and_filter_modules_respects_excluded_modules():
+    assert "InnerBlock" in str(MODULE_LIST)
+    with temp_config_parser(
+        {
+            "service_generation": {
+                "modules": {"excluded": ["00110203-baad-beef-0809-0a0b02dd0e0f"]}
```

Comment:
general usability q as someone who hasn't looked at runtime for a while - are users generally expected to know the guids here?  or how are other config readers supposed to know what module each guid refers to i.e. reading this - how ""00110203-baad-beef-0809-0a0b02dd0e0f" is `InnerBlock` without grepping?

Reply:
A quick fix here might be to change `modules` to `module_guids` so that it's clear these are lists of guids, and leave this open to extension later to add like 
```
module_paths:
    included: ["my_lib.blocks.my_task.some_partial_name*"]
    # or...
    excluded: ["*ESAHierachical*", "my_lib.blocks.some_task.internal.*"]
```

Reply:
Updated to use `module_guids`. Yea I agree with @evaline-ju . Guids arent super readable for usability, I'm unsure of why we chose that when we first started with runtime...

Reply:
Probably because we needed to get things out the door ⏩⏩⏩ and the simplest option was to match guids instead of doing anything more complex